package/src/HostCommunication/HostCommunication.php

```diff
@@ -16,8 +17,13 @@
 
 class HostCommunication
```

Comment:
This class is quite ugly now. It contains my attempts to find a good algorithm. I commented out parts which I don't need now. They might be needed later so I don't want to remove them completely at the moment. I think I'll need to clean the code when   the algorithm'll be ready. It shouldn't be a big problem with all tests